.travis.yml

```diff
@@ -1,31 +1,37 @@
+dist: trusty
 language: php
 matrix:
   fast_finish: true
   include:
     - php: 5.2
+      dist: precise
     - php: 5.3
+      dist: precise
     - php: 5.4
     - php: 5.5
     - php: 5.6
       env: TEST_COVERAGE=1
     - php: 7.0
+    - php: 7.1
+    - php: 7.2
+    - php: nightly
     - php: hhvm
 
+allow_failures:
+  - php: nightly
+
 # Use new container infrastructure
 sudo: false
 
 cache:
   directories:
     - $HOME/.cache/pip
-    - $HOME/.composer/cache
-    - vendor
+    - $HOME/.composer/cache/files
 
 install:
  # Setup the test server
- - phpenv local 5.5
  - composer install --dev --no-interaction
  - TESTPHPBIN=$(phpenv which php)
- - phpenv local --unset
```

Comment:
Does `composer install` work on PHP 5.2 though? That was the issue previously which required the `phpenv` hack.

Reply:
It does not, I'll add back Composer for PHP 5.2 in a follow-up PR, it was quicker and easier to have some CI functionality for the majority of PHP versions rather than nothing, now all but PHP 5.2 and HHVM work, previously all CI jobs were erroring